shell/browser/ui/tray_icon_gtk.cc

```diff
@@ -4,21 +4,62 @@
 
 #include "shell/browser/ui/tray_icon_gtk.h"
 
+#include <gtk/gtk.h>
+
+#include <algorithm>
+
+#include "base/optional.h"
 #include "base/strings/stringprintf.h"
 #include "base/strings/utf_string_conversions.h"
 #include "chrome/browser/ui/views/status_icons/status_icon_linux_dbus.h"
 #include "shell/browser/browser.h"
 #include "shell/common/application_info.h"
 #include "ui/gfx/image/image.h"
+#include "ui/gfx/image/image_skia_operations.h"
 
 namespace electron {
 
 TrayIconGtk::TrayIconGtk() = default;
 
 TrayIconGtk::~TrayIconGtk() = default;
 
+int GetPreferredTrayIconPixelSize() {
+  static base::Optional<int> pixels;
+
+  if (!pixels) {
+    static constexpr GtkIconSize icon_size = GTK_ICON_SIZE_LARGE_TOOLBAR;
+    static constexpr int fallback_px = 22;
```

Comment:
Can you add a comment on how the fallback value is chosen ?

Reply:
Done!

I'm glad you requested that: after some more Googling, I've found the history to not be what I expected. A lot of desktops straight-up coded to an expected height of 22 px, _not_ any GtkIconSize. In my defense, libappindicator's example code does indeed use a GtkIconSize -- but it looks like that works by the coincidence of its values being close to 22 px anyway. I've revised the code to skip GtkIconSize and just pin the height to 22 pixels.

This might not be desirable long-term as it could cause suboptimal icons on high-dpi displays -- but for 8.0.0 it is better than the 8.0.0-beta.X behavior of having 1x1 px tray icons. [This active Chromium issue](https://bugs.chromium.org/p/chromium/issues/detail?id=419673) discusses ongoing upstream work.

More background, if desired, can be found in [this oldish blog post](http://blog.martin-graesslin.com/blog/2014/03/system-tray-in-plasma-next/).

Reply:
@deepak1556 @ckerr This is not desirable at all
![Screenshot_2020-04-27_00-02-33](https://user-images.githubusercontent.com/2256039/80321155-343b0880-881b-11ea-8ac1-da16387c9d64.png)

The icon is disproportionately small. No other application I use has this problem. And I am not using large icons. If I used 48px icons, this would be crazy disproportionate.